### PR TITLE
[NativeAOT] Fix GCDesc computation

### PR DESCRIPTION
**Pull request by SingleAccretion**
Ref: https://github.com/dotnet/runtimelab/pull/2340.

### CODE REVIEW

**Review by VSadov — APPROVED** [review]
LGTM.
Thanks!!

**Comment by MichalStrehovsky** [comment]
Any chance a regression test can be written for this in addition to the one in #86877? GC reporting bugs are always annoying and this could regress again when someone touches this (we clearly have no testing for this case).

**Comment by SingleAccretion** [comment]
> Any chance a regression test can be written for this?

Done. Notably, the existing test already covers exactly this, but only on 32 bit (hence why we only saw fail it downstream).

File: src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/EETypeCreator.cs [hunk 1]
```diff
@@ -565,7 +565,7 @@ private static unsafe int CreateArrayGCDesc(LowLevelList<bool> bitfield, int ran
             {
                 if (numSeries > 0)
                 {
-                    *ptr-- = (short)((bitfield.Count - last) * IntPtr.Size);
+                    *ptr-- = (short)((first + bitfield.Count - last) * IntPtr.Size);
```
Comment (VSadov) [inline src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/EETypeCreator.cs:568]: It looks like this code was broken for MD arrays for a long time. That means we did not have working tests sensitive to this for a while.

Here the original version . The `-2` there hardcodes the base size adjustment for SZ array. For MD array that would need to consider dimensions.

https://github.com/dotnet/runtime/blob/5b62e27e914a0067d85a24545b1013fa8b39b204/src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/EETypeCreator.cs#L568

https://github.com/dotnet/runtime/pull/86877  fixed this case partially, but broke the case where pointers are not at the start of the element. Thus this fix is needed.

**Review by jkotas — APPROVED** [review]
Thank you!

**Pull request merged**